src/fuzzy_match.cc

```diff
@@ -540,18 +543,26 @@ namespace fuzzy
 
         /* let us check the candidates */
         const auto sentence_realtok = _suffixArrayIndex->real_tokens(s_id);
+        const auto cost_upper_bound = lowest_costs.top();
```

Comment:
You should start to restrict with this only when we have already enough matches
`(number_of_matches > 0 && lowest_costs.size() > number_of_matches)`
Same thing for the `continue` that happens du to this part
`cost > cost_upper_bound`

It seems that in some cases (depending on the order of visiting the `pair`) you may lose some secondary matches.

Reply:
We first push `std::numeric_limits<float>::max()` in the heap, which will stay at the `top` as long as we don't have enough matches. With this trick the restriction will only start the first time this condition is true:

```cpp
score < fuzzy || (number_of_matches > 0 && lowest_costs.size() > number_of_matches)
```

So I think the current implementation is correct.

Reply:
Ok, I now see that this is correct.

It is very surprising, can you add a comment
`lowest_costs.push(std::numeric_limits<float>::max());`